lib/srv/desktop/discovery.go

```diff
@@ -66,16 +66,26 @@ func (s *WindowsService) startDesktopDiscovery(ctx context.Context) error {
 
 		GetCurrentResources: func() types.ResourcesWithLabels { return s.lastDiscoveryResults },
 		GetNewResources:     s.getDesktopsFromLDAP,
-		OnCreate:            s.createDesktop,
-		OnUpdate:            s.updateDesktop,
-		OnDelete:            s.deleteDesktop,
-		Log:                 s.cfg.Log,
+		// On the first run of the reconcile loop, a desktop may exist in the database
+		OnCreate: s.createDesktop,
+		OnUpdate: s.updateDesktop,
+		OnDelete: s.deleteDesktop,
+		Log:      s.cfg.Log,
 	})
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	go func() {
+		// initialize lastDiscoverResults with any desktops from a previous run that were saved in the backend
+		initialDesktops, err := s.getDesktopsFromAuthServer(ctx)
+		if err != nil {
+			s.cfg.Log.Errorf("initial fetch of previously discovered desktops from auth server failed : %v", err)
+		}
+
+		s.lastDiscoveryResults = initialDesktops
+		s.cfg.Log.Debugf("desktop discovery starting with initial discovery results of %v", s.lastDiscoveryResults)
```

Comment:
This will likely print a big long blob of text on a system with more than 1 or 2 desktops. Maybe just log how many desktops were found like we do in the reconcile loop.